src/pudl/output_diff/cli.py

```diff
@@ -0,0 +1,403 @@
+            if not fname.endswith(".sqlite"):
+                # This file is not supported by this comparison function
+                continue
+
+            table_diff = self.compare_tables(fname)
+            maybe_append_diff(table_diff)
+            for table_name in table_diff.both:
+                schema_diff = self.compare_table_schemas(fname, table_name)
+                maybe_append_diff(schema_diff)
+                # We could consider comparing tables that have divergent schemas,
+                # but that would be trickier.
+                if schema_diff.sides_equal():
+                    rc_diff = self.compare_table_rowcounts(fname, table_name)
+                    maybe_append_diff(rc_diff)
+                    # if rc_diff.sides_equal():
+                    #    maybe_append_diff(self.compare_table_rows(fname, table_name))
+        return output_diffs
+
+    def compare_tables(self, fname: str) -> ListDiff:
+        """Generates diff comparing the presence/absence of sqlite tables."""
+        baseline_inspector = self.baseline.get_inspector(fname)
+        experiment_inspector = self.experiment.get_inspector(fname)
+
+        baseline_tables = set(baseline_inspector.get_table_names())
+        experiment_tables = set(experiment_inspector.get_table_names())
+        return ListDiff.from_sets(f"{fname}/tables", baseline_tables, experiment_tables)
+
+    def compare_table_schemas(self, fname: str, table_name: str) -> ListDiff:
+        """Generate diff comparing the schema for a given table."""
+        baseline_inspector = self.baseline.get_inspector(fname)
+        experiment_inspector = self.experiment.get_inspector(fname)
+        baseline_cols = [
+            (col["name"], str(col["type"]))
+            for col in baseline_inspector.get_columns(table_name)
+        ]
+        experiment_cols = [
+            (col["name"], str(col["type"]))
+            for col in experiment_inspector.get_columns(table_name)
+        ]
+        return ListDiff.from_sets(
+            f"Schema({fname}/{table_name})",
+            baseline_cols,
+            experiment_cols,
+        )
+
+    def compare_table_rowcounts(self, fname: str, table_name: str) -> RowCountDiff:
+        """Compares the number of rows within given file and table."""
+        logger.info(f"Analyzing rows of {fname}/{table_name}.")
+        brc = self.baseline.get_row_count(fname, table_name)
+        erc = self.experiment.get_row_count(fname, table_name)
+        return RowCountDiff(
+            category=f"RowCount({fname}/{table_name})", left=brc, right=erc
+        )
+
+    def compare_table_rows(self, fname: str, table_name: str) -> RowSampleDiff:
+        """Compares individual rows within given file and table."""
+        # TODO(rousik): this may be very memory/resource expensive for tables
+        # with many rows. We might want to switch this off for very large tables
+        # or switch to some cheaper approach, e.g. statistical sampling or
```

Comment:
Sampling above a configurable threshold seems like a pretty good approach to me!